proto/provenance/msgfees/v1/tx.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+package provenance.msgfees.v1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/provenance-io/provenance/x/msgfees/types";
+
+option java_package        = "io.provenance.msgfees.v1";
+option java_multiple_files = true;
+
+// Msg defines the msgfees Msg service.
+service Msg {
+  // AssessCustomMsgFee endpoint executes the additional fee charges.
+  // This will only emit the event and not persist it to the keeper.  Fees are handled with the custom msg fee handlers
+  // Use Case: smart contracts will be able to charge additional fees and direct partial funds to specified recipient
+  // for executing contracts
+  rpc AssessCustomMsgFee(MsgAssessCustomMsgFeeRequest) returns (MsgAssessCustomMsgFeeResponse);
+}
+
+// MsgAssessCustomMsgFeeRequest defines an sdk.Msg type
+message MsgAssessCustomMsgFeeRequest {
+  option (gogoproto.equal)            = false;
+  option (gogoproto.goproto_getters)  = false;
+  option (gogoproto.goproto_stringer) = false;
+
+  string name = 1; // optional short name for custom msg fee, this will be emitted as a property of the event
+  cosmos.base.v1beta1.Coin amount = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"amount\""
+  ];                    // amount of additional fee that must be paid
+  string recipient = 3; // optional recipient address, the amount is split 50/50 between recipient and fee module. If
+                        // empty, whole amount goes to fee module
+  string from = 4;      // the signer of the msg
```

Comment:
I'm off the opinion to leave this as is.  The from property doesn't really mean it is from that user.  The amount is actually sent from the fee payer of the transaction.  I should actually rename it to signer.  Since, if it is coming from a contract, it will be the contract id when the actual fees are paid from the fee payer.

Reply:
`from` is the canonical name for whom this message is from, it's the primary and first signer that indicates the context the message is processed in. Even in a delegated fee context the from must be an account that holds that permission. This is the correct name for this value.

Reply:
I wasn't talking about the field names, just the field order. But really, my nit about the order now really only leads to a future nit iff the next field to be added is related to the `recipient` field (e.g. customization of the 50/50 split). My nit there would be that that two logically related fields aren't juxtaposed in the field order. And since I can't think of any field that would be added that would be related to the `from`, having it earlier in the ordering probably doesn't lead to this possible future minor annoyance.

But I'm just going to resolve this because it's just me being all sorts of nitty.